dbaccess/acceptanceindex.go

```diff
@@ -0,0 +1,96 @@
+package dbaccess
+
+import (
+	"github.com/kaspanet/kaspad/database2"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/pkg/errors"
+)
+
+var (
+	acceptanceIndexBucket = database2.MakeBucket([]byte("acceptance-index"))
+)
+
+// StoreAcceptanceData stores the given acceptanceData in the database.
+func StoreAcceptanceData(context Context, hash *daghash.Hash, acceptanceData []byte) error {
+	accessor, err := context.accessor()
+	if err != nil {
+		return err
+	}
+
+	key := acceptanceIndexKey(hash)
+	return accessor.Put(key, acceptanceData)
+}
+
+// HasAcceptanceData returns whether the acceptanceData of the given hash
+// has been previously inserted into the database.
+func HasAcceptanceData(context Context, hash *daghash.Hash) (bool, error) {
+	accessor, err := context.accessor()
+	if err != nil {
+		return false, err
+	}
+
+	key := acceptanceIndexKey(hash)
+	return accessor.Has(key)
+}
+
+// FetchAcceptanceData returns the acceptanceData of the given hash.
+// Returns ErrNotFound if the acceptanceData had not been previously
+// inserted into the database.
+func FetchAcceptanceData(context Context, hash *daghash.Hash) ([]byte, error) {
+	accessor, err := context.accessor()
+	if err != nil {
+		return nil, err
+	}
+
+	key := acceptanceIndexKey(hash)
+	acceptanceData, err := accessor.Get(key)
+	if err != nil {
+		if database2.IsNotFoundError(err) {
+			return nil, errors.Wrapf(err, "acceptance data not found for hash %s", hash)
+		}
+		return nil, err
+	}
+
+	return acceptanceData, nil
+}
+
+func acceptanceIndexKey(hash *daghash.Hash) []byte {
+	return acceptanceIndexBucket.Key(hash[:])
+}
+
+// ClearAcceptanceIndex completely removes all acceptanceData entries.
+func ClearAcceptanceIndex() error {
```

Comment:
Maybe Drop instead of Clear, to stay true to the lingo in the flag?
(Or if you don't like Drop for any reason, I'd suggest Remove, both here and in the flag).

Clear sounds a bit like "clear so that you can fill it back up with new data"